themes/theme-ios11/components/FilterBar/index.jsx

```diff
@@ -15,14 +16,20 @@ const { colors, variables: { scroll: { offset = 100 } = {} } } = themeConfig ||
  * @param {Object} props The component props.
  * @returns {JSX}
  */
-function FilterBar({ filters }) {
+function FilterBar({ filters, pattern }) {
   const { hideOnScroll } = useWidgetSettings('@shopgate/engage/components/FilterBar');
   const [active, setActive] = useState(filters !== null && Object.keys(filters).length > 0);
 
   useEffect(() => {
     setActive(filters !== null && Object.keys(filters).length > 0);
   }, [filters]);
 
+  useEffect(() => {
+    if (filters !== null && pattern === CATEGORY_ALL_PATTERN && Object.keys(filters).length === 1) {
+      setActive(false);
```

Comment:
implementing a 2nd effect that invokes the same setter raises some potential issues. 

i didn't verify it, but by reading the code i assume that in case of "category/all" setActive is invoked twice. 1st time with `true` and within the same render step a 2nd time with `false`.

since i didn't give feedback for the whole PR by now i don't know yet if this file still will contain logic related to CATEGORY_ALL_PATTERN in the end, but in that case you should integrate this into the first effect.

Reply:
I reverted the logic to the original approach and added a `sanitizedFilters` variable before where filters with the `isHidden` flag are removed from the object.

The current approach has the issue that `setActive` can only be invoked with `false` at the "CATEGORY_ALL_PATTERN" route, and it can only be invoked with `true` for any other route. But the logic to toggle the FilterBar needs to be independent from the route, but it's related to the state of the filters.